app/views/leads/index.html.erb

```diff
@@ -3,95 +3,94 @@
-    DE - North Rhine-Westphalia (<%= @count_for_northrhine %>)
+    DE - North Rhine-Westphalia (<%= @addresses_count["Germany - North Rhine-Westphalia"] %>)
   </li>
   <li data-filter="Germany - Rhineland-Palatinate">
-    DE - Rhineland-Palatinate (<%= @count_for_rhineland %>)
+    DE - Rhineland-Palatinate (<%= @addresses_count["Germany - Rhineland-Palatinate"] %>)
   </li>
   <li data-filter="Germany - Saarland">
-    DE - Saarland (<%= @count_for_saarland %>)
+    DE - Saarland (<%= @addresses_count["Germany - Saarland"] %>)
   </li>
   <li data-filter="Germany - Saxony">
-    DE - Saxony (<%= @count_for_saxony %>)
+    DE - Saxony (<%= @addresses_count["Germany - Saxony"] %>)
   </li>
   <li data-filter="Germany - Saxony-Anhalt">
-    DE - Saxony-Anhalt (<%= @count_for_s_anhalt %>)
+    DE - Saxony-Anhalt (<%= @addresses_count["Germany - Saxony-Anhalt"] %>)
   </li>
   <li data-filter="Germany - Schleswig-Holstein">
-    DE - Schleswig-Holstein (<%= @count_for_schleswig %>)
+    DE - Schleswig-Holstein (<%= @addresses_count["Germany - Schleswig-Holstein"] %>)
   </li>
   <li data-filter="Germany - Thuringia">
-    DE - Thuringia (<%= @count_for_thuringia %>)
+    DE - Thuringia (<%= @addresses_count["Germany - Thuringia"] %>)
   </li>
   <li data-filter="Greece">
-    Greece (<%= @count_for_greece %>)
+    Greece (<%= @addresses_count["Greece"] %>)
   </li>
   <li data-filter="Turkey">
-    Turkey (<%= @count_for_turkey %>)
+    Turkey (<%= @addresses_count["Turkey"] %>)
   </li>
   <li data-filter="Egypt">
-    Egypt (<%= @count_for_egypt %>)
+    Egypt (<%= @addresses_count["Egypt"] %>)
   </li>
   <li data-filter="Jordan">
-    Jordan (<%= @count_for_jordan %>)
+    Jordan (<%= @addresses_count["Jordan"] %>)
   </li>
   <li data-filter="Lebanon">
-    Lebanon (<%= @count_for_lebanon %>)
+    Lebanon (<%= @addresses_count["Lebanon"] %>)
   </li>
   <li data-filter="Sweden">
-    Sweden (<%= @count_for_sweden %>)
+    Sweden (<%= @addresses_count["Sweden"] %>)
   </li>
   <li data-filter="Norway">
-    Norway (<%= @count_for_norway %>)
+    Norway (<%= @addresses_count["Norway"] %>)
   </li>
   <li data-filter="Findland">
-    Findland (<%= @count_for_findland %>)
+    Findland (<%= @addresses_count["Findland"] %>)
   </li>
   <li data-filter="Austria">
-    Austria (<%= @count_for_austria %>)
+    Austria (<%= @addresses_count["Austria"] %>)
   </li>
   <li data-filter="Switzerland">
-    Switzerland (<%= @count_for_switzerland %>)
+    Switzerland (<%= @addresses_count["Switzerland"] %>)
   </li>
   <li data-filter="France">
-    France (<%= @count_for_france %>)
+    France (<%= @addresses_count["France"] %>)
   </li>
   <li data-filter="United Kingdom">
-    UK (<%= @count_for_uk %>)
+    UK (<%= @addresses_count["United Kingdom"] %>)
   </li>
   <li data-filter="Other">
-    Other (<%= @count_for_other %>)
+    Other (<%= @addresses_count["Other"] %>)
   </li>
 </ol>
 <ul id="tiles" class="wookmark">
-  <% if @leads.present? %>
```

Comment:
This is not necessary, as `Lead.all` would return an empty array if there are no leads, and `[].each` already skips executing anything if it's empty